providers/dns.rb

```diff
@@ -0,0 +1,154 @@
+#
+# Author:: Richard Lavey (richard.lavey@calastone.com)
+# Cookbook Name:: windows
+# Provider:: dns
+#
+# Copyright:: 2015, Calastone Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# See this for info on DNSCMD
+# https://technet.microsoft.com/en-gb/library/cc772069.aspx#BKMK_10
+
+include Chef::Mixin::ShellOut
+include Windows::Helper
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+action :create do
+	if @current_resource.exists
```

Comment:
Could you change \t to two spaces to be consistent with everything else in the cookbook?